### PR TITLE
Add a unit test suite for Animation

### PR DESCRIPTION
**Pull request by Calinou**
This adds unit tests for the Animation *resource*.

Method call, audio and animation playback tracks could be added, but I feel this is a good start already.

### CODE REVIEW

**Review by fire — APPROVED** [review]
@TokageItLab I don't see anything obviously wrong.

File: tests/scene/test_animation.h [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+	CHECK(Math::is_zero_approx(animation->bezier_track_interpolate(0, 0.0)));
+	CHECK(animation->blend_shape_track_interpolate(0, 0.0, nullptr) == ERR_INVALID_PARAMETER);
+	ERR_PRINT_ON;
+}
+
+TEST_CASE("[Animation] Create 3D rotation track") {
+	Ref<Animation> animation = memnew(Animation);
+	const int track_index = animation->add_track(Animation::TYPE_ROTATION_3D);
+	animation->track_set_path(track_index, NodePath("Enemy:rotation"));
+	animation->rotation_track_insert_key(track_index, 0.0, Quaternion(Vector3(0, 1, 2)));
```
Comment (fire) [inline tests/scene/test_animation.h:143]: This is odd. Isn't this supposed to go to basis first?

Comment (Calinou) [reply tests/scene/test_animation.h:143]: In `master`, this creates a Quaternion from an Euler angle.

**Comment by fire** [comment]
There's a test failure.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**